src/vm/readytoruninfo.cpp

```diff
@@ -890,6 +890,84 @@ PCODE ReadyToRunInfo::MethodIterator::GetMethodStartAddress()
     return ret;
 }
 
+BOOL ReadyToRunInfo::GenericMethodIterator::Next()
+{
+    m_current = m_enum.GetNext();
+    return !m_current.IsNull();
+}
+
+MethodDesc *ReadyToRunInfo::GenericMethodIterator::GetMethodDesc_NoRestore()
+{
+    _ASSERTE(!m_current.IsNull());
+
+    PCCOR_SIGNATURE pBlob = (PCCOR_SIGNATURE)m_current.GetBlob();
```

Comment:
Nit: The general convention I see in this file is to not declare all locals at the begining of the function, but only when needed. It would be nice if you could also do that here.

Reply:
It's certainly possible, I was just trying to have to avoid an `if (FAILED(hr))` block after each call to SigParser